runtime/nft_test.go

```diff
@@ -999,13 +999,13 @@ pub contract TopShotShardedCollection {
         }
 
         // getIDs returns an array of the IDs that are in the Collection
-        pub fun getIDs(): [UInt64] {
+        view pub fun getIDs(): [UInt64] {
 
             var ids: [UInt64] = []
             // Concatenate IDs in all the Collections
-            for key in self.collections.keys {
+            for i, key in self.collections.keys {
                 for id in self.collections[key]?.getIDs() ?? [] {
-                    ids.append(id)
+                    ids[i] = id
```

Comment:
This will result in an index out-of-bounds error. Why was it changed?

Reply:
This was changed in order to make this function `view`, as `append` is a non-view function and thus not callable in a `view` context. However, if this doesn't need to be `view` then we can change it back.

Reply:
This is a limitation of the system really; because we can only annotate a function as being `view` or not `view`, there is no additional information given about what kind of mutating operation might happen inside of a non-`view` function like we would have with a true effect system. We have no way right now to say that a non-`view` method on an array only mutates the receiver, for example, so we can't express the idea that `append` is safe to use in `view` contexts when the receiver is local to that context. In order to be able to model this later, we would need to add more complexity to `view` annotations, or an additional purity keyword, to allow us to describe which kinds of mutations a function is allowed to perform, which feels like trouble than it's worth.

Reply:
That makes sense, thank you for the explanation 👌